src/jacktrip_globals.h

```diff
@@ -99,6 +99,8 @@ const int gWaitCounter = 60;
 /// \name Network related ports
 //@{
 const int gDefaultPort = 4464; ///< Default JackTrip Port
+const int gBindPortLow = 3464; ///< lowest Bindport
+const int gBindPortHigh = 5464; ///< highest Bindport
```

Comment:
Do we need to restrict the port range this heavily? Wouldn't it work to just check that we have a valid, non-zero, unsigned 16 bit integer? (We could also prevent the use of well known ports below 1024 if we need to.)

Reply:
I recall signed wrap-around being a problem, not sure if there are others @cchafe ?

BTW, I'm curious why the client sends the port it's bound to in the protocol, versus just using the port from the UDP header? Seems likely this is what causes NAT issues with JackTrip. Namely, every device behind a NAT needs to be configured to listen on a different port. Also it seems like we are just getting lucky that NAT's aren't remapping this to another random port when they forward packets?

Reply:
There were some uninitialized variables in the same neighborhood. Fixed those that I saw and made the types consistent. The port range was chosen arbitrarily. Essentially, it's doing the job a firewall would do. Historically, TCP / UDP were the same but became independent to support a wider range of ephemeral ports + the ability to have multiple clients behind the same NAT. We just had a situation today where excluding all but a single TCP port was advantageous for "securing" a session (definitely a hack). A future feature would be setting that from cmd line,

Reply:
The only problem I see with arbitrarily limiting the port range is that we're not mirroring that restriction in Settings.cpp, which could cause a lot of confusion when people try using ports outside of that narrowly defined range on the client side.

On the NAT front, yes, we probably should be overriding the port that's sent by the client with the port that's taken from the initial UDP packet, and it's a trivial, three line, fix. (At which point, the restriction on the sent port becomes kind of moot except as a very basic security device.) I've just implemented it in QJackTrip, but the UDP hub code in that has diverged quite a bit from the current dev branch.

Reply:
Scratch that. It's not a trivial fix. On a networking level we're already using the port in the UDP header rather than the client supplied one. We're using the port sent by the client to identify whether a client is unique or not, and to replace the existing connection if we already have a record of it. There's a less trivial fix for it (having JackTripWorker notify UdpHubListener of the actual port being used and updating the record), but it's going to be a bit of a concurrency headache when multiple machines from behind the same NAT initiate a connection on the same port at the same time.